services/scheduler/scheduler.go

```diff
@@ -298,32 +308,92 @@ func (s *Service) convertDBTask(dbTask *models.ScheduledTask) (Task, error) {
 	switch dbTask.Type {
 	case models.ScheduledMySQLBackupTask:
 		data := dbTask.Data.MySQLBackupTask
-		params := CommonBackupTaskParams{
-			ServiceID:     data.ServiceID,
-			LocationID:    data.LocationID,
-			Name:          data.Name,
-			Description:   data.Description,
-			Retention:     data.Retention,
-			Retries:       data.Retries,
-			RetryInterval: data.RetryInterval,
+		task = &mySQLBackupTask{
+			common: common{
+				id: dbTask.ID,
+			},
+			BackupTaskParams: &BackupTaskParams{
+				ServiceID:     data.ServiceID,
+				LocationID:    data.LocationID,
+				Name:          data.Name,
+				Description:   data.Description,
+				DataModel:     data.DataModel,
+				Mode:          data.Mode,
+				Retention:     data.Retention,
+				Retries:       data.Retries,
+				RetryInterval: data.RetryInterval,
+			},
 		}
-		task = NewMySQLBackupTask(s.backupService, params)
 	case models.ScheduledMongoDBBackupTask:
 		data := dbTask.Data.MongoDBBackupTask
-		params := CommonBackupTaskParams{
-			ServiceID:     data.ServiceID,
-			LocationID:    data.LocationID,
-			Name:          data.Name,
-			Description:   data.Description,
-			Retention:     data.Retention,
-			Retries:       data.Retries,
-			RetryInterval: data.RetryInterval,
+		task = &mongoDBBackupTask{
+			common: common{
+				id: dbTask.ID,
+			},
+			BackupTaskParams: &BackupTaskParams{
+				ServiceID:     data.ServiceID,
+				LocationID:    data.LocationID,
+				Name:          data.Name,
+				Description:   data.Description,
+				DataModel:     data.DataModel,
+				Mode:          data.Mode,
+				Retention:     data.Retention,
+				Retries:       data.Retries,
+				RetryInterval: data.RetryInterval,
+			},
 		}
-		task = NewMongoBackupTask(s.backupService, params)
+
 	default:
-		return task, errors.Errorf("unknown task type: %s", dbTask.Type)
+		return nil, errors.Errorf("unknown task type: %s", dbTask.Type)
 	}
 
-	task.SetID(dbTask.ID)
 	return task, nil
 }
+
+func checkPreconditions(q *reform.Querier, data *models.ScheduledTaskData, enabled bool, scheduledTaskID string) error {
+	switch {
+	case data.MySQLBackupTask != nil:
+	case data.MongoDBBackupTask != nil:
+		data := data.MongoDBBackupTask
+		if enabled {
+			return checkMongoDBBackupPreconditions(q, data.Mode, data.ServiceID, scheduledTaskID)
+		}
+	}
+	return nil
+}
+
+func checkMongoDBBackupPreconditions(q *reform.Querier, mode models.BackupMode, serviceID, scheduleID string) error {
+	switch mode {
+	case models.PITR:
+		// PITR backup can be enabled only if there is no other scheduled backups.
+		tasks, err := models.FindScheduledTasks(q, models.ScheduledTasksFilter{
+			Disabled:  pointer.ToBool(false),
+			ServiceID: serviceID,
+		})
+		if err != nil {
+			return err
+		}
+
+		for _, task := range tasks {
+			if task.ID != scheduleID {
+				return status.Error(codes.FailedPrecondition, "Scheduled PITR backup can be enabled only if there is no other scheduled backups enabled.")
```

Comment:
Enabling can be confused with `toggling something on/off`. I suggest we use the word `create` or `register`, i.e.
```suggestion
				return status.Error(codes.FailedPrecondition, "A scheduled PITR backup can be created only if there are no other scheduled backups.")
```

Reply:
Done partially. This check is about enabling, because user can create multiple disabled PITR backups